main.go

```diff
@@ -10,22 +10,28 @@ import (
 	"github.com/okkur/reposeed/cmd/reposeed/config"
 )
 
+const SupportedConfigVersion = "v1"
+
 func main() {
 	err := godotenv.Load()
 	if err != nil {
 		log.Println("Couldn't find .env file. Reading environment variables from system")
 	}
 	app := gin.Default()
 	config := &config.Config{}
-	app.POST("/generate", func(ctx *gin.Context) {
-		ctx.BindJSON(config)
-		filename, err := generator.CreateFiles(*config, config.Project.Name, os.Getenv("STORAGE"))
-		if err.Code != 200 {
-			ctx.JSON(400, err)
-			ctx.Abort()
-		} else {
-			ctx.File(filename)
-		}
-	})
-	app.Run(os.Getenv("PORT"))
+	if config.Project.Version == SupportedConfigVersion {
+		app.POST("/generate", func(ctx *gin.Context) {
+			ctx.BindJSON(config)
+			filename, err := generator.CreateFiles(*config, config.Project.Name, os.Getenv("STORAGE"))
+			if err.Code != 200 {
+				ctx.JSON(400, err)
+				ctx.Abort()
+			} else {
+				ctx.File(filename)
+			}
+		})
+		app.Run(os.Getenv("PORT"))
```

Comment:
app.Run() runs a web server listening on the given port. It should be outside of any statement and at the end of the main function.

Reply:
Thanks for the helpful overview - I'll make these changes!